### PR TITLE
[@mantine/core] Image: fix possible permanent placeholder after hydration

### LINKED ISSUE #629: Race condition between hydration and image load event
### What package has an issue

@mantine/core

### Describe the bug

When an Image with placeholder is server-side rendered, the component displayed in browser can be hydrated after load event and in this case will keep displaying placeholder.

### In which browser did the problem occur

Any

### If possible, please include a link to a codesandbox with the reproduced problem

_No response_

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Looking at this issue's comment https://github.com/facebook/react/issues/15446#issuecomment-484709466

I could implement a fix for Image with a useEffect

```typescript
useEffect(() => {
  const { current } = internalImgRef;
  if (current?.complete) {
    setError(current.naturalHeight === 0);
    setLoaded(current.naturalHeight !== 0);
  }
}, [src])
```

### PR DESCRIPTION
**Pull request by Stouffi**
This PR fixes #629

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v14.9.0
+v14.17.0
```
Comment (rtivital) [inline .nvmrc:1]: Is there any reason to change node version?

**Comment by rtivital** [comment]
Thanks for a PR, have you tested it to work in ssr environment, e.g. in [next template](https://github.com/mantinedev/mantine-next-template)?

File: .nvmrc (hunk 1 above)
Comment (Stouffi) [reply .nvmrc:1]: I was stuck with an error stating that a dependency required node >= 14.17 during my first `yarn install`

Comment (rtivital) [reply .nvmrc:1]: Alright, then let's leave it in 14.17 then

**Comment by Stouffi** [comment]
@rtivital I encountered the issue with my stack which is viteJS + SSR and with the fix, the problem with placeholder disappeared.

**Comment by rtivital** [comment]
Okay, I'll run some more tests with it then

**Comment by Stouffi** [comment]
Oh btw, did you mean testing **manually** with other SSR environment? I couldn't figure out how to add a test with an html string and an hydration using `@testing-library/react` in `src/mantine-core/src/components/Image/Image.test.tsx`

**Comment by rtivital** [comment]
Yes, I do not think it is possible to check it within jest-dom environment, so it should be checked manually at least in Next.js

**Comment by rtivital** [comment]
I've checked with next, it works fine

**Pull request merged**